spec/controllers/api/v1/assignment_events_controller_spec.rb

```diff
@@ -30,13 +30,20 @@
     end
 
     it "noops if a conflicting assignment already exists" do
-      FactoryBot.create(:assignment, visitor: visitor, split: split, variant: "control")
+      assignment = nil
+      Timecop.freeze(1.year.ago) do
+        assignment = FactoryBot.create(:assignment, visitor: visitor, split: split, variant: "control")
+      end
+
+      assignment.reload
+      original_attributes = assignment.attributes
 
       expect {
         post :create, params: create_params
       }.not_to change { PreviousAssignment.count }
```

Comment:
We might be able to extend this to also test that we don't change the assignment attributes. Something like:
```
expect {
  post :create, params: create_params
}.not_to change { PreviousAssignment.count }
.and not_change { assignment.reload.attributes }
```

Though that might also mean you would need to create the negated `not_change` matcher, assuming the TT code base doesn't already have it.

So take it or leave it! Obviously not a big deal! On my first read I just missed why we were saving the `original_attributes`

Reply:
seems a little fancy. I'm OK with not understanding why you need original_attributes until they're checked.